logpyle/runalyzer.py

```diff
@@ -353,11 +359,15 @@ def execute_magic(self, cmdline):
             from pylab import title
             title(args)
         elif cmd == "plot":
+            cursor = self.db.db.execute(self.db.mangle_sql(args))
+            columnnames = [column[0] for column in cursor.description]
             self.db.plot_cursor(self.db.db.execute(
-                self.db.mangle_sql(args)))
+                self.db.mangle_sql(args)), columnnames)
         elif cmd == "scatter":
+            cursor = self.db.db.execute(self.db.mangle_sql(args))
+            columnnames = [column[0] for column in cursor.description]
             self.db.scatter_cursor(self.db.db.execute(
-                self.db.mangle_sql(args)))
+                self.db.mangle_sql(args)), columnnames)
```

Comment:
These run the query twice, which seems wasteful. Why?

Reply:
Fixed.